### PR TITLE
[logstash] Add fullnameOverride setting

### PR DESCRIPTION
**Pull request by morganchristiansson**
Added fullnameOverride to logstash by copy pasting from kibana/filebeat/metricbeat charts.

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by morganchristiansson** [comment]
This is to allow overriding the name of the logstash service and just generally better naming options than "logstash-logstash" and "logstash-main-main" that can be set by nameOverride value today.

**Comment by jmlrt** [comment]
jenkins test this please

File: logstash/tests/logstash_test.py [hunk 1]
```diff
@@ -571,3 +571,22 @@ def test_adding_a_service():
...
         'name': 'beats', 'port': 5044, 'protocol': 'TCP', 'targetPort': 5044}
+
+def test_setting_fullnameOverride():
+    config = '''
+fullnameOverride: 'logstash-custom'
+'''
+    r = helm_template(config)
+
+    custom_name = 'logstash-custom'
+    assert custom_name in r['daemonset']
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py]: Logstash is using a statefulset and not a daemonset.

Comment (morganchristiansson) [reply logstash/tests/logstash_test.py]: Fixed. I did not attempt to run the tests locally thanks for triggering the Jenkins build.

**Comment by dorkvodor** [comment]
Hiya! any updates on devops-ci check?

**Comment by dorkvodor** [comment]
Hiya! Can this be merged now?

**Comment by morganchristiansson** [comment]
I managed to run the tests locally and created PR #463 to fix make pytest.

I updated tests to remove checks that are not applicable for default logstash deployment and it's passing locally now.

Please re-run the jenkins tests

**Comment by jmlrt** [comment]
jenkins test this please

File: logstash/tests/logstash_test.py [hunk 2]
```diff
@@ -579,14 +579,6 @@ def test_setting_fullnameOverride():
...
-    assert {
-               'name': 'data',
-               'hostPath': {
-                   'path': '/var/lib/' + custom_name + '-default-data',
-                   'type': 'DirectoryOrCreate'
-               }
-           } in volumes
+    assert custom_name in r['statefulset']
+    assert r['statefulset'][custom_name]['spec']['template']['spec']['containers'][0]['name'] == 'logstash'
+    volumes = r['statefulset'][custom_name]['spec']['template']['spec']['volumes']
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py]: The last line `volumes = ...` doesn't seem to be used

Comment (morganchristiansson) [reply logstash/tests/logstash_test.py]: Where I copied it from it was used for this check
```
    assert {
               'name': 'data',
               'hostPath': {
                   'path': '/var/lib/' + custom_name + '-default-data',
                   'type': 'DirectoryOrCreate'
               }
           } in volumes
```
I don't think we care about custom_name being in volume mount path... So I'm going to delete this.

Comment (morganchristiansson) [reply logstash/tests/logstash_test.py]: I rebased  my branch on upstream/master, amended this change into the _Update tests_ commit and force-pushed

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by morganchristiansson** [comment]
Thanks! I assume this will be released in the next elastic-stack release? Current release is 7.5.2 so  7.5.3 and later will include this fix?

**Comment by jmlrt** [comment]
Yes, this will be released in the next one which should be 7.5.3 or 7.6.0.

**Pull request merged**